src/CrcParameters.h

```diff
@@ -0,0 +1,309 @@
+#define CRC8_SAEJ1850_ZERO_REF_OUT  false
+
+#define CRC8_8H2F_POLYNOME          0x2F
+#define CRC8_8H2F_INITIAL           0xFF
+#define CRC8_8H2F_XOR_OUT           0xFF
+#define CRC8_8H2F_REF_IN            false
+#define CRC8_8H2F_REF_OUT           false
+
+#define CRC8_WCDMA_POLYNOME         0x9B
+#define CRC8_WCDMA_INITIAL          0xFF
+#define CRC8_WCDMA_XOR_OUT          0x00
+#define CRC8_WCDMA_REF_IN           false
+#define CRC8_WCDMA_REF_OUT          false
+
+#define CRC8_DARC_POLYNOME          0x39
+#define CRC8_DARC_INITIAL           0x00
+#define CRC8_DARC_XOR_OUT           0x00
+#define CRC8_DARC_REF_IN            true
+#define CRC8_DARC_REF_OUT           true
+
+#define CRC8_DVB_S2_POLYNOME        0xD5
+#define CRC8_DVB_S2_INITIAL         0x00
+#define CRC8_DVB_S2_XOR_OUT         0x00
+#define CRC8_DVB_S2_REF_IN          false
+#define CRC8_DVB_S2_REF_OUT         false
+
+#define CRC8_EBU_POLYNOME           0x1D
+#define CRC8_EBU_INITIAL            0xFF
+#define CRC8_EBU_XOR_OUT            0x00
+#define CRC8_EBU_REF_IN             true
+#define CRC8_EBU_REF_OUT            true
+
+#define CRC8_ICODE_POLYNOME         0x1D
+#define CRC8_ICODE_INITIAL          0xFD
+#define CRC8_ICODE_XOR_OUT          0x00
+#define CRC8_ICODE_REF_IN           false
+#define CRC8_ICODE_REF_OUT          false
+
+#define CRC8_ITU_POLYNOME           0x07
+#define CRC8_ITU_INITIAL            0x00
+#define CRC8_ITU_XOR_OUT            0x55
+#define CRC8_ITU_REF_IN             false
+#define CRC8_ITU_REF_OUT            false
+
+#define CRC8_DALLAS_MAXIM_POLYNOME  0x31 // oneWire
+#define CRC8_DALLAS_MAXIM_INITIAL   0x00
+#define CRC8_DALLAS_MAXIM_XOR_OUT   0x00
+#define CRC8_DALLAS_MAXIM_REF_IN    true
+#define CRC8_DALLAS_MAXIM_REF_OUT   true
+
+#define CRC8_ROHC_POLYNOME          0x07
+#define CRC8_ROHC_INITIAL           0xFF
+#define CRC8_ROHC_XOR_OUT           0x00
+#define CRC8_ROHC_REF_IN            true
+#define CRC8_ROHC_REF_OUT           true
+
+#define CRC8_AUTOSAR_POLYNOME       0x2F
+#define CRC8_BLUETOOTH_POLYNOME     0xA7
+#define CRC8_CCITT_POLYNOME         0x07
+#define CRC8_GSM_B_POLYNOME         0x49
+
+
+// CRC 12
+#define CRC12_POLYNOME              0x080D // x12 + x3 + x2 + 1 =  0000 1000 0000 1101 = 0x80D
+#define CRC12_INITIAL               0x0000
+#define CRC12_XOR_OUT               0x0000
+#define CRC12_REF_IN                false
+#define CRC12_REF_OUT               false
+
+#define CRC12_CCITT_POLYNOME        0x080F
+#define CRC12_CDMA2000_POLYNOME     0x0F13
+#define CRC12_GSM_POLYNOME          0x0D31
+
+
+// CRC 16
+#define CRC16_POLYNOME              0x8001 // x15 + 1 =  1000 0000 0000 0001 = 0x8001
+#define CRC16_INITIAL               0x0000
+#define CRC16_XOR_OUT               0x0000
+#define CRC16_REF_IN                false
+#define CRC16_REF_OUT               false
+
+#define CRC16_CCITT_POLYNOME        0x1021 // x13 + X5 + 1 =  0001 0000 0010 0001 = 0x1021
+#define CRC16_CCITT_INITIAL         0x0000
```

Comment:
About this one I'm not sure, before you have ` 0xFFFF` initial value for that, but in calculators `CRC16_CCITT_FALSE_INITIAL` has this value. Change it to `0xFFFF` and remove `CRC16_CCITT_FALSE_..` defines